src/aws-fsx/ontap-file-system.ts

```diff
@@ -0,0 +1,514 @@
+    ) {
+      return;
+    }
+    if (
+      automaticBackupRetention.toMilliseconds() < Duration.days(1).toMilliseconds() ||
+      automaticBackupRetention.toDays() > 90
+    ) {
+      throw new Error('automaticBackupRetention must be between 1 and 90 days or be equal to 0');
+    }
+  }
+
+  private validateDailyAutomaticBackupStartTime(
+    automaticBackupRetention?: Duration,
+    dailyAutomaticBackupStartTime?: aws_fsx.DailyAutomaticBackupStartTime,
+  ): void {
+    if (!dailyAutomaticBackupStartTime) {
+      return;
+    }
+
+    const automaticBackupDisabled = !automaticBackupRetention || automaticBackupRetention?.toDays() === 0;
+
+    if (dailyAutomaticBackupStartTime && automaticBackupDisabled) {
+      throw new Error(
+        "'automaticBackupRetention' period must be set a non-zero day when 'dailyAutomaticBackupStartTime' is set",
+      );
+    }
+  }
+
+  private validateDiskIops(storageCapacityGiB: number, diskIops?: number, haPairs: number = 1): void {
+    if (
+      diskIops == null ||
+      Token.isUnresolved(diskIops) ||
+      Token.isUnresolved(storageCapacityGiB) ||
+      Token.isUnresolved(haPairs)
+    ) {
+      return;
+    }
+
+    const minDiskIops = storageCapacityGiB * 3 * haPairs;
+    const maxDiskIops = 200_000 * haPairs;
+
+    if (!Number.isInteger(diskIops) || diskIops < minDiskIops || diskIops > maxDiskIops) {
+      throw new Error(`\'diskIops\' must be an integer between ${minDiskIops} and ${maxDiskIops}, got ${diskIops}`);
+    }
+  }
+
+  private validateEndpointIpAddressRange(deploymentType: OntapDeploymentType, endpointIpAddressRange?: string): void {
+    if (endpointIpAddressRange == null || Token.isUnresolved(endpointIpAddressRange)) {
+      return;
+    }
+    if (deploymentType !== OntapDeploymentType.MULTI_AZ_1 && deploymentType !== OntapDeploymentType.MULTI_AZ_2) {
+      throw new Error("'endpointIpAddressRange' can only be specified for multi-AZ file systems");
+    }
+    if (!/^[^\u0000\u0085\u2028\u2029\r\n]{9,17}$/.test(endpointIpAddressRange)) {
+      throw new Error(
+        "'endpointIpAddressRange' must be between 9 and 17 characters long and not contain any of the following characters: \\u0000, \\u0085, \\u2028, \\u2029, \\r, or \\n",
+      );
+    }
+  }
+
+  private validateFsxAdminPassword(fsxAdminPassword?: string): void {
+    if (fsxAdminPassword == null || Token.isUnresolved(fsxAdminPassword)) {
+      return;
+    }
+    if (!/^[^\u0000\u0085\u2028\u2029\r\n]{8,50}$/.test(fsxAdminPassword)) {
+      throw new Error(
+        "'fsxAdminPassword' must be between 8 and 50 characters long and not contain any of the following characters: \\u0000, \\u0085, \\u2028, \\u2029, \\r, or \\n",
+      );
+    }
+    // must contain at least one English letter and one number, and must not contain the word 'admin'.
+    if (!/[a-zA-Z]/.test(fsxAdminPassword) || !/\d/.test(fsxAdminPassword) || /admin/i.test(fsxAdminPassword)) {
```

Comment:
These rules are not documented but I failed to deploy in these condition. Therefore, I added this validation.

https://dev.classmethod.jp/articles/fsxn-password-cannot-be-used-within-the-last-six-times/

Reply:
I hadn't realized that... Cloudformation document says 50...

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-fsx-filesystem-ontapconfiguration.html#cfn-fsx-filesystem-ontapconfiguration-fsxadminpassword

Which value should be accept?

Reply:
This would no longer be necessary: https://github.com/open-constructs/aws-cdk-library/pull/42#discussion_r1781414058